src/core/ir/eval.rs

```diff
@@ -150,3 +166,72 @@ impl IR {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    mod merge {
```

Comment:
Move tests closer to implementation or vice versa.

Reply:
I moved it closer to implementation

Reply:
It's already along the implementation. Please, note, it's a test for a `IR::Merge` not merge_right or anything